app/configurator/configurator-state.tsx

```diff
@@ -261,7 +272,7 @@ export const applyDimensionToFilters = (
     } else {
       // Add filter for this dim if it's not one of the selected multi filter fields, but
       // only when the dimension isn't grouped, as only then we need to switch to a single filter.
-      if (isHidden && !isGrouped) {
+      if ((isHidden || !isField) && !isGrouped) {
```

Comment:
Nit: I find this condition quite hard to understand 😅 Is there a way to make it more clear or at least update the comment to describe the cases that are covered? But maybe it's just me 👴